superset/db_engine_specs/base.py

```diff
@@ -933,6 +933,9 @@ def execute(cls, cursor: Any, query: str, **kwargs: Any) -> None:
         :param kwargs: kwargs to be passed to cursor.execute()
         :return:
         """
+        if not cls.allows_sql_comments:
+            query = sql_parse.ParsedQuery(query).strip_comments()
```

Comment:
I seem to remember that parsing the SQL statements is computationally intensive - would it make sense to add this to the `process_statement` method instead to avoid an additional parsing pass on the query?

Reply:
I checked the description of `SQL_QUERY_MUTATOR`, and it only mentions that it's typically used to add comments - but not strictly restricted to that. If we want to block mutation from happening, we should rename the property to `allows_query_mutator` or similar to highlight that we're blocking that from happening rather than just stripping comments.

As this is only affecting a single engine, I feel this is the least invasive solution for now - we can always change this later if the overhead becomes an issue.

Reply:
This needs to happen before any query execution on engines that don't support comments. Just instantiating `ParsedQuery` is much less intensive now, but yes some overhead exist.
We used to have default comments on SQLab for example, probably was considered safe since comments are an ANSI SQL 92 standard. `allows_query_mutator` is a good option but seems more evasive

I've made a feature request on opendistro SQL:
https://github.com/opendistro-for-elasticsearch/sql/issues/1051

^^^ Community support is awesome on opendistro SQL, and they are heavily developing and about to release a major change.

We can make a quick char check before instantiating `ParsedQuery`, not pretty, but would avoid unnecessary processing on all statements

Reply:
I support the idea of doing a quick char check, best of both worlds 👍  Also, I'm assuming this is a temporary fix, so we should be able to disable it when/if your request is implemented.